util/GameRules.cpp

```diff
@@ -136,6 +137,18 @@ std::map<std::string, std::string> GameRules::GetRulesAsStrings() {
     return retval;
 }
 
+std::vector<const GameRule*> GameRules::GetSortedByCategoryAndRank() {
+    CheckPendingGameRules();
+    std::vector<const GameRule*> sorted_rules(m_game_rules.size(), nullptr);
+    std::transform(m_game_rules.begin(), m_game_rules.end(), sorted_rules.begin(), [](const auto& rule_pair) { return &rule_pair.second; });
+    std::sort(sorted_rules.begin(), sorted_rules.end(), [](const GameRule* lhs, const GameRule* rhs) {
+        auto lhs_category_rank = GameRuleCategories::GetCategoryRank(lhs->category);
```

Comment:
is this the only place you use this function? If so, it doesn't need to be declared in a separate header and implemented in a spearate .cpp. that would also get rid of your linker errors.

Reply:
ah hmm currently yes. Suppose I wanted to stick it explicitly with definition of categories but could be hidden behind/within GameRules.cpp indeed. 

I've pushed change to cmakelists to fix the linking error but may as well do as you suggest